url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/model/mxGraphModel.java

```diff
@@ -0,0 +1,2572 @@
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#getEdgeAt(Object, int)
+	 */
+	public Object getEdgeAt(Object parent, int index)
+	{
+		return (parent instanceof mxICell) ? ((mxICell) parent)
+				.getEdgeAt(index) : null;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#isVertex(Object)
+	 */
+	public boolean isVertex(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).isVertex() : false;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#isEdge(Object)
+	 */
+	public boolean isEdge(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).isEdge() : false;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#isConnectable(Object)
+	 */
+	public boolean isConnectable(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).isConnectable()
+				: true;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#getValue(Object)
+	 */
+	public Object getValue(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).getValue() : null;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#setValue(Object, Object)
+	 */
+	public Object setValue(Object cell, Object value)
+	{
+		execute(new mxValueChange(this, cell, value));
+
+		return value;
+	}
+
+	/**
+	 * Inner callback to update the user object of the given mxCell
+	 * using mxCell.setValue and return the previous value,
+	 * that is, the return value of mxCell.getValue.
+	 */
+	protected Object valueForCellChanged(Object cell, Object value)
+	{
+		Object oldValue = ((mxICell) cell).getValue();
+		((mxICell) cell).setValue(value);
+
+		return oldValue;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#getGeometry(Object)
+	 */
+	public mxGeometry getGeometry(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).getGeometry()
+				: null;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#setGeometry(Object, mxGeometry)
+	 */
+	public mxGeometry setGeometry(Object cell, mxGeometry geometry)
+	{
+		if (geometry != getGeometry(cell))
+		{
+			execute(new mxGeometryChange(this, cell, geometry));
+		}
+
+		return geometry;
+	}
+
+	/**
+	 * Inner callback to update the mxGeometry of the given mxCell using
+	 * mxCell.setGeometry and return the previous mxGeometry.
+	 */
+	protected mxGeometry geometryForCellChanged(Object cell, mxGeometry geometry)
+	{
+		mxGeometry previous = getGeometry(cell);
+		((mxICell) cell).setGeometry(geometry);
+
+		return previous;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#getStyle(Object)
+	 */
+	public String getStyle(Object cell)
+	{
+		return (cell instanceof mxICell) ? ((mxICell) cell).getStyle() : null;
+	}
+
+	/* (non-Javadoc)
+	 * @see com.mxgraph.model.mxIGraphModel#setStyle(Object, String)
+	 */
+	public String setStyle(Object cell, String style)
+	{
+		if (style != getStyle(cell))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.